mm2src/coins/utxo/utxo_tests.rs

```diff
@@ -4256,3 +4256,98 @@ fn test_utxo_validate_valid_and_invalid_pubkey() {
     assert!(coin.validate_other_pubkey(&[1u8; 20]).is_err());
     assert!(coin.validate_other_pubkey(&[1u8; 8]).is_err());
 }
+
+#[test]
+fn test_block_header_utxo_loop() {
+    use crate::utxo::utxo_builder::{block_header_utxo_loop, BlockHeaderUtxoLoopExtraArgs};
+    use futures::future::{Either, FutureExt};
+    use keys::hash::H256 as H256Json;
+
+    static mut CURRENT_BLOCK_COUNT: u64 = 13;
+
+    ElectrumClient::get_block_count
+        .mock_safe(move |_| MockResult::Return(Box::new(futures01::future::ok(unsafe { CURRENT_BLOCK_COUNT }))));
+    let expected_steps: Arc<Mutex<Vec<(u64, u64)>>> = Arc::new(Mutex::new(vec![]));
+    let expected_steps_to_move = expected_steps.clone();
+
+    ElectrumClient::retrieve_headers.mock_safe(move |this, from, count| {
+        let (expected_from, expected_count) = expected_steps_to_move.lock().unwrap().remove(0);
+        assert_eq!(from, expected_from);
+        assert_eq!(count, expected_count);
+        MockResult::Continue((this, from, count))
+    });
```

Comment:
Since `ElectrumClient::retrieve_headers` takes `from` and `to` values, could you please change this to
```rust
ElectrumClient::retrieve_headers.mock_safe(move |this, from, to| {
        let (expected_from, expected_to) = expected_steps_to_move.lock().unwrap().remove(0);
        assert_eq!(from, expected_from);
        assert_eq!(count, expected_to);
        MockResult::Continue((this, from, to))
    });
```